plugins/modules/digital_ocean_database.py

```diff
@@ -0,0 +1,347 @@
+        if not json_data:
+            json_data = self.get_by_name(self.module.params['name'])
+        return json_data
+
+    def ensure_online(self, database_id):
+        end_time = time.time() + self.wait_timeout
+        while time.time() < end_time:
+            response = self.rest.get('databases/{0}'.format(database_id))
+            json_data = response.json
+            database = json_data.get('database', None)
+            if database is not None:
+                status = database.get('status', None)
+                if status is not None:
+                    if status == 'online':
+                        return json_data
+            time.sleep(10)
+        self.module.fail_json(msg='Waiting for database online timeout')
+
+    def create(self):
+        json_data = self.get_database()
+
+        if json_data is not None:
+            database = json_data.get('database', None)
+            if database is not None:
+                self.module.exit_json(changed=False, data=json_data)
+            else:
+                self.module.fail_json(changed=False, msg='Unexpected error, please file a bug')
+
+        if self.module.check_mode:
+            self.module.exit_json(changed=True)
+
+        request_params = dict(self.module.params)
+        del request_params['id']
+
+        response = self.rest.post('databases', data=request_params)
+        json_data = response.json
+        if response.status_code >= 400:
+            self.module.fail_json(changed=False, msg=json_data['message'])
+        database = json_data.get('database', None)
+        if database is None:
+            self.module.fail_json(changed=False, msg='Unexpected error, please file a bug')
+        if self.wait:
+            database_id = database.get('id', None)
+            if database_id is None:
+                self.module.fail_json(changed=False, msg='Unexpected error, please file a bug')
+            json_data = self.ensure_online(database_id)
+        self.module.exit_json(changed=True, data=json_data)
+
+    def delete(self):
+        json_data = self.get_database()
+        if json_data is not None:
+            if self.module.check_mode:
+                self.module.exit_json(changed=True)
+            database = json_data.get('database', None)
+            database_id = database.get('id', None)
+            database_name = database.get('name', None)
+            database_region = database.get('region', None)
+            if database_id is not None:
+                response = self.rest.delete('databases/{0}'.format(database_id))
+                json_data = response.json
+                if response.status_code == 204:
+                    self.module.exit_json(changed=True, msg='Deleted database {0} ({1}) in {2}'.format(database_name, database_id, database_region))
+                self.module.fail_json(changed=False,
+                                      msg='Failed to delete database {0} ({1}) in {2}: {3}'
```

Comment:
Consider saying `in region` in this log message in case users aren't familiar with DigitalOcean's region slugs. (there are several occurrences of this in this file; this is just the first one I noticed)